internal/state/state.go

```diff
@@ -240,13 +240,21 @@ func (state State) MakeBlock(
 	// Build base block with block data.
 	block := types.MakeBlock(height, txs, lastCommit, evidence)
 
-	// TODO(@glnro): Update logic once merged with #2205
-	// Check if PBTS Enabled: state.ConsensusParams.PBTS.PBTSEnabled(height)
+	// Set time.
+	var timestamp time.Time
+	switch {
+	case state.ConsensusParams.Feature.PbtsEnabled(height):
+		timestamp = time.Now()
+	case height == state.InitialHeight:
```

Comment:
Shouldn't this condition come before?

Reply:
If using PBTS from genesis, then the genesis time will not be timely, as normally the genesis will be created long before the chain is started the genesis timestamp will be well in the past.